src/main/java/org/apache/commons/collections4/bloomfilter/SimpleHasher.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.collections4.bloomfilter;
+
+import java.util.Objects;
+import java.util.function.IntPredicate;
+
+/**
+ * A Hasher that implements combinatorial hashing as as described by
+ * <a href="https://www.eecs.harvard.edu/~michaelm/postscripts/tr-02-05.pdf">Krisch and Mitzenmacher</a>.
+ * <p>
+ * Common use for this hasher is to generate a byte array as the output of a hashing
+ * or MessageDigest algorithm.</p>
+ *
+ * @since 4.5
+ */
+public final class SimpleHasher implements Hasher {
+    /**
+     * A default increment used when the requested increment is zero. This is the same
+     * default increment used in Java's SplittableRandom random number generator.  It is the
+     * fractional representation of the golden ratio (0.618...) with a base of 2^64.
+     */
+    public static final long DEFAULT_INCREMENT = 0x9e3779b97f4a7c15L;
+
+    /**
+     * The initial hash value.
+     */
+    private final long initial;
+
+    /**
+     * The value to increment the hash value by.
+     */
+    private final long increment;
+
+    /**
+     * Convert bytes to long.
+     * @param byteArray the byte array to extract the values from.
+     * @param offset the offset to start extraction from.
+     * @param len the length of the extraction, may be longer than 8.
+     * @return
+     */
+    private static long toLong(byte[] byteArray, int offset, int len) {
+        long val = 0;
+        len = Math.min(len, Long.BYTES);
+        for (int i = 0; i < len; i++) {
+            val <<= 8;
+            val |= (byteArray[offset + i] & 0x00FF);
+        }
+        return val;
+    }
+
+    /**
+     * Constructs the SimpleHasher from a byte array.
+     * <p>The byte array is split in 2 and each half is interpreted as a long value.
+     * Excess bytes are ignored.  This simplifies the conversion from a Digest or hasher algorithm output
+     * to the two values used by the SimpleHasher.</p>
+     * <p><em>If the second long is zero the DEFAULT_INCREMENT is used instead.</em></p>
+     * @param buffer the buffer to extract the longs from.
+     * @throws IllegalArgumentException is buffer length is zero.
+     */
+    public SimpleHasher(byte[] buffer) {
+        if (buffer.length == 0) {
+            throw new IllegalArgumentException("buffer length must be greater than 0");
+        }
+        int segment = buffer.length / 2;
+        this.initial = toLong(buffer, 0, segment);
+        long possibleIncrement = toLong(buffer, segment, buffer.length - segment);
+        this.increment = possibleIncrement == 0 ? DEFAULT_INCREMENT : possibleIncrement;
+    }
+
+    /**
+     * Constructs the SimpleHasher from 2 longs.  The long values will be interpreted as unsigned values.
```

Comment:
These are currently not interpreted as unsigned values. Either change this, or update the mod method to compute the result of Long.unsignedReminder:
```Java
static int mod(long dividend, int divisor) {
    // See Hacker's Delight (2nd ed), section 9.3.
    // Assume divisor is positive.
    // Divide half the unsigned number and then double the quotient result.
    final long quotient = ((dividend >>> 1) / divisor) << 1;
    final long remainder = dividend - quotient * divisor;
    // remainder in [0, 2 * divisor)
    return (int) (remainder >= divisor ? remainder - divisor : remainder);
}
```

I prefer this method if only because you can document it as behaving functionally identically to Long.unsignedRemainder, test it as such and then the SimpleHasher becomes more closely aligned with Krisch and Mitzenmacher's source method using unsigned 64-bit integers.

The effect on speed would be negligible.